spec/javascripts/ga_cross_domain_form_tracking_spec.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-env jasmine, jquery */
+/* global attachCrossDomainTrackerToInput */
+
+describe('attachCrossDomainTrackerToInput', function () {
+  var form
+  var tracker
+  var linker
+
+  beforeEach(function () {
+    form = $('<form method="POST" action="/somewhere" class="account-signup">' +
+                '<input type="hidden" name="key" value="value" />' +
+                '<button type="submit">Create a GOV.UK account</button>' +
+              '</form>')
+
+    window.ga = {
+      getAll: function () {}
+    }
+
+    window.gaplugins = {
+      Linker: function () {}
+    }
+
+    linker = {
+      decorate: function () {}
+    }
+
+    spyOn(window.ga, 'getAll').and.returnValue([])
+    spyOn(window.gaplugins, 'Linker').and.returnValue(linker)
+    spyOn(linker, 'decorate').and.returnValue('/somewhere?_ga=abc123')
+  })
+
+  afterEach(function () {
+    form.remove()
+  })
+
+  it('leaves the form action unchanged if ga is not present', function () {
+    tracker = []
+    attachCrossDomainTrackerToInput(form, tracker)
+    expect(form.attr('action')).toEqual('/somewhere')
+  })
+
+  it('leaves the form action unchanged there are no trackers in ga', function () {
+    tracker = []
+    attachCrossDomainTrackerToInput(form, tracker)
+    expect(form.attr('action')).toEqual('/somewhere')
+  })
+
+  it('modifies the form action to append ids from ga to the destination url', function () {
+    tracker = [{ ga_mock: 'foobar' }]
```

Comment:
This might not need any changes, but I don't really understand this. Is it the case that if `tracker` contains anything, the code runs? Or does it specifically have to contain a `ga_mock` attribute? If so, what is `ga_mock`?

Reply:
So this code will only be fired if this array has any length.
In the wild we get two fairly complex ga tracking objects, with the first being cross domain.

For these tests, as we're mocking the results (again starting to feel a bit more like testing the mocks than the logic, but I suppose we're intercepting known steps, so perhaps ok?).

So i just need something here and i figured `[{}]` unclear and not descriptive enough.

So the keys and values don't do anything except describe that this is an object. Happy to take any suggestions on preferred ways of doing things